subsys/net/lib/sockets/sockets.c

```diff
@@ -119,6 +119,12 @@ int _impl_zsock_socket(int family, int type, int proto)
 	}
 #endif
 
+#if defined(CONFIG_NET_SOCKETS_RAW)
+	if (type == SOCK_RAW) {
```

Comment:
Per the revelation in https://github.com/zephyrproject-rtos/zephyr/pull/12564#issuecomment-458646644 (challenges to it are welcome), this should dispatch on `family == AF_PACKET` instead.

Reply:
Done.